src/utils/commits/index.ts

```diff
@@ -1 +1,19 @@
-export * from "./bucketByCommit";
+import { FileDiffsFragment } from "gql/generated/types";
+
+export { bucketByCommit } from "./bucketByCommit";
+
+/**
+ * Indicates if a user created a patch with the --preserve-commits flag.
+ * We can tell that the user opted to preserve commits if the description of a fileDiff is non-empty,
+ * as it will contain the associated commit message.
+ * @param {FileDiffsFragment[]} fileDiffs - array containing the fileDiffs for a particular commit
+ * @return {boolean} true if commits are preserved, false if otherwise
+ */
+export const shouldPreserveCommits = (
+  fileDiffs: FileDiffsFragment[]
+): boolean => {
+  if (fileDiffs.length && fileDiffs[0].description !== "") {
+    return true;
+  }
+  return false;
+};
```

Comment:
This should probably live somewhere by enqueue modal i think its only used by that. Unless you think it might be used elsewhere.

Reply:
It's used by `CodeChanges.tsx`!